core/test/element_logical_operations_test.cpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+#include <gtest/gtest.h>
+
+#include <Eigen/Geometry>
+
+#include "scipp/common/constants.h"
+#include "scipp/core/element/logical_operations.h"
+#include "scipp/units/unit.h"
+
+#include "fix_typed_test_suite_warnings.h"
+
+using namespace scipp;
+using namespace scipp::core::element;
+
+template <typename T> class ElementLessTest : public ::testing::Test {};
+template <typename T> class ElementGreaterTest : public ::testing::Test {};
+template <typename T> class ElementLessEqualTest : public ::testing::Test {};
+template <typename T> class ElementGreaterEqualTest : public ::testing::Test {};
+template <typename T> class ElementEqualTest : public ::testing::Test {};
+template <typename T> class ElementNotEqualTest : public ::testing::Test {};
+using ElementLessTestTypes = ::testing::Types<double, float, int64_t, int32_t>;
+TYPED_TEST_SUITE(ElementLessTest, ElementLessTestTypes);
+TYPED_TEST_SUITE(ElementGreaterTest, ElementLessTestTypes);
+TYPED_TEST_SUITE(ElementLessEqualTest, ElementLessTestTypes);
+TYPED_TEST_SUITE(ElementGreaterEqualTest, ElementLessTestTypes);
+TYPED_TEST_SUITE(ElementEqualTest, ElementLessTestTypes);
+TYPED_TEST_SUITE(ElementNotEqualTest, ElementLessTestTypes);
+
+TYPED_TEST(ElementLessTest, unit) {
+  const units::Unit m(units::m);
+  EXPECT_EQ(less(m, m), units::dimensionless);
+  const units::Unit rad(units::rad);
+  EXPECT_THROW(less(rad, m), except::UnitError);
+}
+
+TYPED_TEST(ElementLessTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(less(y, x), true);
+  x = -1;
+  EXPECT_EQ(less(y, x), false);
+}
+
+template <int T, typename Op> bool is_no_variance_arg() {
+  return std::is_base_of_v<core::transform_flags::expect_no_variance_arg_t<T>,
+                           Op>;
+}
+
+TYPED_TEST(ElementLessTest, value_only_arguments) {
+  using Op = decltype(less);
+  EXPECT_TRUE((is_no_variance_arg<0, Op>())) << " y has variance ";
+  EXPECT_TRUE((is_no_variance_arg<1, Op>())) << " x has variance ";
+}
+
+TYPED_TEST(ElementGreaterTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(greater(y, x), false);
+  x = -1;
+  EXPECT_EQ(greater(y, x), true);
+}
+
+TYPED_TEST(ElementLessEqualTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(less_equal(y, x), true);
+  x = 1;
+  EXPECT_EQ(less_equal(y, x), true);
+}
+
+TYPED_TEST(ElementGreaterEqualTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(greater_equal(y, x), false);
+  x = 1;
+  EXPECT_EQ(greater_equal(y, x), true);
+}
+
+TYPED_TEST(ElementEqualTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(equal(y, x), false);
+  x = 1;
+  EXPECT_EQ(equal(y, x), true);
+}
+
+TYPED_TEST(ElementNotEqualTest, value) {
+  using T = TypeParam;
+  T y = 1;
+  T x = 2;
+  EXPECT_EQ(not_equal(y, x), true);
+  x = 1;
+  EXPECT_EQ(not_equal(y, x), false);
```

Comment:
Don't we need *three* assertions (instead of two) in every test case (here and above) to safely distinuish the operators? For example, this test would also pass if `not_equal` would accidentially be implemented using `<` due to a hypothetical copy-and-paste error in the header.